compiler/src/dotty/tools/dotc/core/Substituters.scala

```diff
@@ -153,7 +153,10 @@ object Substituters:
         if (tp.binder == from) to(tp.paramNum) else tp
       case tp: NamedType =>
         if (tp.prefix `eq` NoPrefix) tp
-        else tp.derivedSelect(substParams(tp.prefix, from, to, theMap))
+        else
+          val pre1 = substParams(tp.prefix, from, to, theMap)
+          if pre1.isValueTypeOrWildcard || (pre1 eq NoPrefix) then tp.derivedSelect(pre1)
+          else tp
```

Comment:
Maybe return an ErrorType if this can only happen in case of errors? Also couldn't the check be in the implementation of derivedSelect?